spec/classes/mod/passenger_spec.rb

```diff
@@ -2,6 +2,185 @@
 
 describe 'apache::mod::passenger', :type => :class do
   it_behaves_like "a mod class, without including apache"
+  context "validating all passenger params - using Debian" do
+    let :facts do
+      {
+          :osfamily               => 'Debian',
+          :operatingsystemrelease => '6',
+          :kernel                 => 'Linux',
+          :concat_basedir         => '/dne',
+          :lsbdistcodename        => 'squeeze',
+          :operatingsystem        => 'Debian',
+          :id                     => 'root',
+          :path                   => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+          :is_pe                  => false,
+      }
+    end
+    it { is_expected.to contain_class("apache::params") }
+    it { is_expected.to contain_apache__mod('passenger') }
+    it { is_expected.to contain_package("libapache2-mod-passenger") }
+    it { is_expected.to contain_file('zpassenger.load').with({
+                                                                 'path' => '/etc/apache2/mods-available/zpassenger.load',
+                                                             }) }
+    it { is_expected.to contain_file('passenger.conf').with({
+                                                                'path' => '/etc/apache2/mods-available/passenger.conf',
+                                                            }) }
+
+    passenger_config_options = {
+        'passenger_allow_encoded_slashes' => {Type: 'OnOff', PassOpt: :PassengerAllowEncodedSlashes},
```

Comment:
It looks like you're using capitalized key names whereas in ruby this is not canonical. 

e.g. `Type` would be `type`, `PassOpt` would be `pass_opt`

Also, is there a particular reason the values for PassOpt are symbols instead of strings?